### PR TITLE
PDF Upload enhancements

### LINKED ISSUE #1217: PDF upload enhancements
Desired enhancements:
* pressing enter will submit the split PDF item
* pressing tab only goes between user id input/submit/delete
* pop-out button for full PDF in the table view
* autofill for user ids

### PR DESCRIPTION
**Pull request by wangj35**
- Closes #1217.
- Fixed bug where sometimes the split item PDF could not be found or submitted.
- Pressing enter scrolls down to the next input as well.
- Bulk PDFs with spaces in the name can now be submitted.
- Autofill for student submission is all students, autofill for split PDF items is students who don't have any submissions yet.

### CODE REVIEW

File: site/app/views/submission/HomeworkView.php [hunk 1]
```diff
@@ -57,10 +57,32 @@ public function showGradeable($gradeable, $days_late) {
         $current_version_number = $gradeable->getCurrentVersionNumber();
         $return = <<<HTML
 <script type="text/javascript" src="{$this->core->getConfig()->getBaseUrl()}js/drag-and-drop.js"></script>
+<script src="https://code.jquery.com/jquery-1.12.4.js"></script>
+<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
```
Comment (MasterOdin) [inline site/app/views/submission/HomeworkView.php]: We already import both jquery and jquery-ui (https://github.com/Submitty/Submitty/blob/master/site/app/views/GlobalView.php#L58-L59), so not sure why you're doing this here?

File: site/app/views/submission/HomeworkView.php [hunk 2]
```diff
@@ -97,6 +119,8 @@ public function showGradeable($gradeable, $days_late) {
     <script type="text/javascript">
         $(document).ready(function() {
             var cookie = document.cookie;
+            student_ids = $student_ids;
```
Comment (MasterOdin) [inline site/app/views/submission/HomeworkView.php]: If you're using PHP functions, please use the `{$student_ids}` just to keep it consistent through the code base.

File: site/app/views/submission/HomeworkView.php [hunk 3]
```diff
@@ -444,23 +496,30 @@ function makeSubmission(user_id, highest_version, is_pdf, path, count) {
                     foreach ($files as $filename => $details) {
                         $clean_timestamp = str_replace("_", " ", $timestamp);
                         $path = $details["path"];
-                        if (strpos($filename, 'cover') == false) {
-                            // add each file that is not a cover to count_array 
-                            // each entry is in format timestamp/filename
-                            $count_array[$count] = $timestamp."/".$filename;
+                        if (strpos($filename, "cover") == false) {
```
Comment (MasterOdin) [inline site/app/views/submission/HomeworkView.php]: use `=== false` here as strpos can return `false` (string not found) or `0` (string at beginning) which are both `== false`.

File: site/cgi-bin/pdf_check.cgi [hunk 4]
```diff
@@ -83,8 +83,8 @@ try:
         for j in range(0,div):
             out_pdf = filename[:-4] + "_" + str(j) + ".pdf"
             out_cover_pdf = filename[:-4] + "_" + str(j) + "_cover.pdf"
-            subprocess.call("pdftk {in_pdf} cat {start}-{stop} output {out_pdf}".format(in_pdf=filename,start=j*num+1,stop=(j+1)*num,out_pdf=out_pdf),shell=True)
-            subprocess.call("pdftk {in_pdf} cat {start} output {out_pdf}".format(in_pdf=filename,start=j*num+1,out_pdf=out_cover_pdf),shell=True)
+            subprocess.call("pdftk \"{in_pdf}\" cat {start}-{stop} output \"{out_pdf}\"".format(in_pdf=filename,start=j*num+1,stop=(j+1)*num,out_pdf=out_pdf),shell=True)
+            subprocess.call("pdftk \"{in_pdf}\" cat {start} output \"{out_pdf}\"".format(in_pdf=filename,start=j*num+1,out_pdf=out_cover_pdf),shell=True)
```
Comment (MasterOdin) [inline site/cgi-bin/pdf_check.cgi]: Don't use shell=True in `subprocess` calls unless you absolutely need to. Use `subprocess.call(['pdftk', filename, 'cat', ...])`. Apply above to `subprocess.check_output`

**Comment by holzbh** [comment]
Functionality seems fine but when I split a PDF (16 pages) and specify 2 pages per PDF, the previews I see are only 1 page (the first page of the split PDFs). If I open the actual files they are correctly 2 pages, but it's weird not being able to see the full preview. Was that by design?

**Pull request merged**